openedx/core/djangoapps/content_tagging/api.py

```diff
@@ -160,19 +145,32 @@ def tag_content_object(
 
     Raises ValueError if the proposed tags are invalid for this taxonomy.
     Preserves existing (valid) tags, adds new (valid) tags, and removes omitted (or invalid) tags.
+
+    We require that this taxonomy is linked to the content object's "org" or linked to "all orgs" (None).
     """
+    object_key: UsageKey | CourseKey
+
     if not taxonomy.system_defined:
         # We require that this taxonomy is linked to the content object's "org" or linked to "all orgs" (None):
+        try:
+            object_key = UsageKey.from_string(object_id)
+        except InvalidKeyError:
+            try:
+                object_key = CourseKey.from_string(object_id)
```

Comment:
Should use `LearningContextKey` (not `CourseKey`) and `UsageKey` like the [previous code](https://github.com/openedx/edx-platform/pull/34146/files#diff-7d88f29d73182315ff6d741f5997d4223f6303acad8d0fdf6d9bdd4e0a70eaedL101-L104) -- this supports content library keys too.

Reply:
I made a comment below about this

Reply:
> I think we should only use CourseKey | UsageKey (because we rely on the org info to tag content)

Ah ok, agreed. I didn't realise LibraryLocator inherited from CourseKey, that's cool.

Reply:
Do you think we should create a task to fix this through the code?

Reply:
@rpenido I think this is too minor to warrant a task.. but if you're working on affected code, feel free to change it as you go, and cite this discussion.